File: Daily Challanges/2026-06-June/2026-06-29.js
function getMood(genre, bpm) {
  if (bpm >= 60 && bpm <= 109 && genre === "classical") {
    return "focus";
  } else if (bpm >= 60 && bpm <= 89 && genre === "electronic") {
    return "focus";
  } else if (bpm >= 60 && bpm <= 180 && genre === "pop") {
    return "happy";
  } else if (bpm >= 110 && bpm <= 180 && genre === "classical") {
    return "happy";
  } else if (bpm >= 60 && bpm <= 129 && genre === "rock") {
    return "happy";
  } else if (bpm >= 90 && bpm <= 134 && genre === "electronic") {
    return "happy";
  } else if (bpm >= 130 && bpm <= 180 && genre === "rock") {
    return "hype";
  } else if (bpm >= 135 && bpm <= 180 && genre === "electronic") {
    return "hype";
  }

  return "Unknown";
}

console.log(getMood("rock", 111));
console.log(getMood("electronic", 74));
console.log(getMood("classical", 180));
console.log(getMood("rock", 155));
console.log(getMood("electronic", 90));
console.log(getMood("classical", 67));
console.log(getMood("pop", 100));
console.log(getMood("electronic", 135));

File: Daily Challanges/2026-06-June/2026-06-29.py
def get_mood(genre, bpm):
    if 60 <= bpm <= 109 and genre == "classical":
        return "focus"
    elif 60 <= bpm <= 89 and genre == "electronic":
        return "focus"
    elif 60 <= bpm <= 180 and genre == "pop":
        return "happy"
    elif 110 <= bpm <= 180 and genre == "classical":
        return "happy"
    elif 60 <= bpm <= 129 and genre == "rock":
        return "happy"
    elif 90 <= bpm <= 134 and genre == "electronic":
        return "happy"
    elif 130 <= bpm <= 180 and genre == "rock":
        return "hype"
    elif 135 <= bpm <= 180 and genre == "electronic":
        return "hype"
    return "Unknown"


print(get_mood("rock", 111))
print(get_mood("electronic", 74))
print(get_mood("classical", 180))
print(get_mood("rock", 155))
print(get_mood("electronic", 90))
print(get_mood("classical", 67))
print(get_mood("pop", 100))
print(get_mood("electronic", 135))
